builtin/providers/aws/resource_aws_rds_cluster.go

```diff
@@ -499,6 +511,16 @@ func resourceAwsRDSClusterRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("reader_endpoint", dbc.ReaderEndpoint)
 	d.Set("replication_source_identifier", dbc.ReplicationSourceIdentifier)
 
+	var roles []string
+	for _, r := range dbc.AssociatedRoles {
+		if *r.Status == "ACTIVE" {
+			roles = append(roles, *r.RoleArn)
```

Comment:
The role status is one of `ACTIVE`, `PENDING`, or `INVALID`; do you not think it's worth storing the role even if it's not in the right state? I worry something will be `INVALID` and the user won't see it, and then Terraform will think it's just not there and try to reapply it.